solar_system.rb

```diff
@@ -0,0 +1,58 @@
+class SolarSystem
+  attr_reader :star_name, :planets
+
+  def initialize(star_name)
+    @star_name = star_name
+    @planets = []
+  end
+
+  # Create a method add_planet,
+  #which will take an instance of Planet as a parameter and add it to the list of planets.
+  def add_planet(planet) #Planet.name
+    #iterate through each Planet.name and put it in the @planets array.
+    @planets << planet
+  end
+
+  def list_planets
+    result = "Planets orbitting:\n"
+    planet_names = @planets.map do |planet|
+      planet.name
+    end
+    planet_names.each_with_index do |name, index|
+      result += "#{index + 1}. #{name}\n"
+    end
+    return result
+    # @planets.name specify that you're searching the name
+  end
+
+  def list_planets_one
+    result = "Planets orbitting:\n"
+    @planets.each_with_index do |planet, index|
+      result += "#{index + 1}. #{planet.name}\n"
+    end
+    return result
+  end
+
+  def find_planet_by_name_two(name)
+    return @planets.find do |planet|
+             planet.name.downcase == name.downcase
+           end
+  end
+
+  def new_planet
```

Comment:
This would be better as a method which takes a `Planet` as an argument and adds it to the `planets` array.  That way you're separating the user-input task from the task of storing and organizing the planets.